Update README's (#40)
viash-hub/rnaseq: README.qmd

### Modular design
The workflow is built in a modular fashion, where most of the base functionality is provided by components from [biobox](https://www.viash-hub.com/packages/biobox/latest), supplemented by custom base components and workflow components in this package. This architecture ensures both flexibility and reproducibility while leveraging established bioinformatics tools.

### Standardized components
Each of the workflow components is implemented as a stand-alone module with a standardized interface. This design philosophy offers several advantages:

1. Isolation of Tools and Functionality: Each subworkflow as well as its individual components can be executed independently as stand-alone entities when you need only specific functionality. 

2. Consistent Interfaces: All components follow standardized input/output conventions, making it easy to connect or replace them. 

3. Reusability: Each component or workflow can be integrated seamlessly as a dependency in another workflow. 

### Workflow Structure

> The [main workflow](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/rnaseq) has 6 sub-workflows that can also be run independently.

Reviewer: Just 'workflow'? 🤔 
```suggestion
The [workflow](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/rnaseq) has 6 sub-workflows that can also be run independently.
```

4. [Post-processing](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/post_processing): This is a workflow for duplicate read marking (picard MarkDuplicates), transcript assembly and quantification (StringTie), and creation of bigWig coverage files.

5. [Pseudo alignment and quantification](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/pseudo_alignment_and_quant): This is a workflow for performing pseudo alignment and transcript quantification using Salmon or Kallisto. 

6. [Final QC](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/quality_control): This is a workflow for performing extensive quality control (RSeQC, dupRadar, Qualimap, Preseq, DESeq2, featureCounts). It presents QC for raw reads, alignments, gene biotype, sample similarity, and strand specificity (MultiQC).

> 

Reviewer: There is a lot repetition of 'This is a workflow' here

> ### Executing individual workflow steps

Reviewer: I think this is a neat feature of viash, but I am not sure how relevant this is for this project (its relatively high up in the readme, but I am not sure it is that important)

I am also thinking about the maintenance of this readme when we move to newer versions of dependencies.

* [`samtools/samtools_sort`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/samtools/samtools_sort)
* [`samtools/samtools_index`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/samtools/samtools_index)
* [`samtools/samtools_stats`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/samtools/samtools_stats)
* [`samtools/samtools_flagstat`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/samtools/samtools_flagstat)
* [`samtools/samtools_idxstats`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/samtools/samtools_idxstats)
* [`multiqc`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/multiqc) (work in progress - updating `assets/multiqc_config.yaml`)
* [`fastp`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/fastp) (work in progress)
* [`rsem/rsem_prepare_reference`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/rsem/rsem_prepare_reference) (work in progress)
* [`rsem/rsem_calculate_expression`](https://www.viash-hub.com/packages/biobox/v0.3.0/components/rsem/rsem_calculate_expression) (work in progress)

> ## Example usage

Reviewer: Maybe a list of options (CLI, nextflow executable and seqera cloud) can be shown here, so that the reader knows that there are different options for executors before needing to scroll down? My first instinct was that there was only the nextflow executable option because that one came first

To get started, we need to:

> 1.  Install [`nextflow`](https://www.nextflow.io/docs/latest/getstarted.html) system-wide

Reviewer: System wide is not really a requirement. 
```suggestion
1.  Install [`nextflow`](https://www.nextflow.io/docs/latest/getstarted.html)
```

> 1.  Install [`nextflow`](https://www.nextflow.io/docs/latest/getstarted.html) system-wide

Reviewer: Is this required when using seqera cloud?

2.  Fetch the test data:

``` bash
bin/minimal_test.sh
bin/get_minimal_test_data.sh
```

## Run the Workflow

> The main rnaseq workflow is available via [Viash Hub](https://viash.io/viash-hub/rnaseq.vsh/), where you can receive instructions on how to run the main workflow as well as individual subworkflows or components.

Reviewer: This link is dead.

> The main rnaseq workflow is available via [Viash Hub](https://viash.io/viash-hub/rnaseq.vsh/), where you can receive instructions on how to run the main workflow as well as individual subworkflows or components.

Reviewer: Maybe?
```suggestion
This package is available via [Viash Hub](https://viash.io/viash-hub/rnaseq.vsh/), where you can receive instructions on how to run the main workflow as well as individual subworkflows or components.
```

To run the main workflow, browse to the [rnaseq](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/rnaseq) workflow on Viash Hub. Here you can find an overview on the formats of the input and output files, as well as a detailed list of required and optional parameters to run the workflow.

> ### Run using Nextflow

Reviewer: Should we specify that this example will only work on Linux and requires some familiarity with the cli?

After clicking launch, we can now follow the instructions on screen.

![](assets/launch_workflow.png)

> 1. In the first step, we choose our execution environment. In this example, we choose Nextflow.

Reviewer: ```suggestion
1.  The options for the package, version and and component fields have already been populated. The first step is to select the execution environment, which is Nextflow in this example. Click 'Next'
```

![](assets/nextflow_execution.png)

> 2. We can now fill in the parameters for the workflow. In this example, we choose the test data we downloaded earlier. We select the `advanced form` option, to be able to process multiple samples in parallel.

Reviewer: ```suggestion
2. We can now fill in the parameters for the workflow. In this example, use the locations of the test data we downloaded earlier. We select the `advanced form` option, to be able to process multiple samples in parallel.
```

We fill out the global parameters first - those are the parameters that apply to all samples.

![](assets/global_params.png)

Then, we fill in our parameter sets - this is one parameter set for each samples. Note that each sample can consist of multiple fastq files!

![](assets/parameter_set_1.png)
![](assets/parameter_set_2.png)

3. Once we hit launch, we can execute the workflow by following the instructions on the screen:

``` bash
cat > params.yaml <<'EOM'
param_list:
  - id: "WT_REP1"
    fastq_1: [ "SRR6357070_1.fastq.gz", "SRR6357071_1.fastq.gz" ]
    fastq_2: [ "SRR6357070_2.fastq.gz", "SRR6357071_2.fastq.gz" ]
    strandedness: "reverse"
  - id: "WT_REP2"
    fastq_1: [ "SRR6357072_1.fastq.gz" ]
    fastq_2: [ "SRR6357072_2.fastq.gz" ]
    strandedness: "reverse"
fasta: "testData/minimal_test/reference/genome.fasta"
publish_dir: "full_pipeline_test/"
gtf: "testData/minimal_test/reference/genes.gtf.gz"
transcript_fasta: "testData/minimal_test/reference/transcriptome.fasta"
EOM

nextflow run https://packages.viash-hub.com/vsh/rnaseq.git \
  -revision v0.2.0 \
  -main-script target/nextflow/workflows/rnaseq/main.nf \
  -params-file params.yaml \
  -latest \
  -resume
```

> ### Run using Seqera Cloud

Reviewer: Should this one come first?

It's also possible to run the workflow directly on [Seqera Cloud](https://cloud.seqera.io/). The required [Nextflow schema files](https://nextflow-io.github.io/nf-schema/latest/nextflow_schema/nextflow_schema_specification/) are provided with the workflows. Since Seqera Cloud does not support multiple-value parameters when using the form-based input, we will use Viash Hub to launch the [workflow](https://www.viash-hub.com/packages/rnaseq/v0.2.0/components/workflows/rnaseq). 

1. First, we need to create an API token for your Seqera Cloud account.
2. Next, we can launch the workflow by selecting `Seqera Cloud` as execution environment. Here you can add your API key, as well as the Workspace ID and Compute Environment. 

![](assets/seqera_cloud_execution.png)

> 3. We can now fill in the parameters, as described in step 2 under `Run using Nextflow`. Note that the test data need to reside on a cloud bucket for Seqera Cloud execution.

Reviewer: Would it not be possible to use the github URLs for the test date here with nextflow?

4. By launching the workflow via Viash Hub, it will be executed on Seqera Cloud in your workspace environment of choice.

### Run using the CLI

To run the workflow as an executable using the CLI, [Viash](https://viash.io/installation/) needs to be installed.

1. On Viash Hub, we can select `executable` as Execution Environment on Viash Hub.

> ![](assets/cli_execution.png)

Reviewer: This is a bit of a weird image here, as it higlights seqera cloud?

> 2. We fill in the parameters, as described in step 2 under `Run using Nextflow`.

Reviewer: I think it is possible to create links between sections.
```suggestion
2. We fill in the parameters, as described in step 2 under `Run using Nextflow`.
```

3. When launching the workflow, we receive instruction on how to run the workflow natively on your local machine via the CLI:

```bash
viash run rnaseq@v0.2.0/workflows/rnaseq -- \
  --fasta testData/minimal_test/reference/genome.fasta \
  --gtf testData/minimal_test/reference/genes.gtf.gz \
  --transcript_fasta testData/minimal_test/reference/transcriptome.fasta \
  --id WT_REP1 \
  --fastq_1 "SRR6357070_1.fastq.gz;SRR6357071_1.fastq.gz" \
  --fastq_2 "SRR6357070_2.fastq.gz;SRR6357071_2.fastq.gz" \
  --strandedness reverse\

>   ---engine native

Reviewer: Does this work already? 🤔
